app/controllers/course/controller.rb

```diff
@@ -9,16 +9,24 @@ class Course::Controller < ApplicationController
   #
   #   {
   #      title: 'Sidebar Item Title'
+  #      type: :admin # Will be treat as :normal if not set
+  #      path: path_to_the_component
   #      unread: 0 # or nil
   #   }
   #
   # The elements are rendered on all Course controller subclasses as part of a nested template.
-  def sidebar
+  # @param [Symbol] type The type of sidebar item, all sidebar items will be returned if the type
+  # is not specified
+  # @return [Array] The array of sidebar items of the given type
+  def sidebar(type = nil)
```

Comment:
Have updated here as what you suggested.

Now it will return all sidebar items when type is not given. And if an item does not set its type, it will be considered as `:normal`